projects/RabbitMQ.Client/client/events/EventingBasicConsumer.cs

```diff
@@ -86,10 +87,12 @@ public override void HandleBasicConsumeOk(string consumerTag)
         /// </remarks>
         public override void HandleBasicDeliver(string consumerTag, ulong deliveryTag, bool redelivered, string exchange, string routingKey, in ReadOnlyBasicProperties properties, ReadOnlyMemory<byte> body)
         {
-            base.HandleBasicDeliver(consumerTag, deliveryTag, redelivered, exchange, routingKey, properties, body);
-            Received?.Invoke(
-                this,
-                new BasicDeliverEventArgs(consumerTag, deliveryTag, redelivered, exchange, routingKey, properties, body));
+            BasicDeliverEventArgs eventArgs = new BasicDeliverEventArgs(consumerTag, deliveryTag, redelivered, exchange, routingKey, properties, body);
```

Comment:
I should change this to not create the EventArgs and skip an allocation if tracing isn't enabled.